### PR TITLE
add benchmarking warp against python itk

### LINKED ISSUE #3691: Registration benchmarking - Warp
**Is your feature request related to a problem? Please describe.**
Though unit-tested the `Warp` layer is not compared to established repositories yet.

**Describe the solution you'd like**
- write a Jupyter Notebook benchmarking resampling-related features both qualitatively and quantitatively against SimpleITK and python itk
- add python-itk-related quantitative results to unit-testing 

**Describe alternatives you've considered**
N/A

**Additional context**
[python itk example](https://itk.org/ITKExamples/src/Filtering/ImageGrid/WarpAnImageUsingADeformationField/Documentation.html)
[Simple ITK example](http://insightsoftwareconsortium.github.io/SimpleITK-Notebooks/Python_html/21_Transforms_and_Resampling.html)

### PR DESCRIPTION
**Pull request by kate-sann5100**
Signed-off-by: kate-sann5100 <yiwen.li@st-annes.ox.ac.uk>

Fixes #3691.

### Description
Add python-itk-related quantitative results to unit-testing.

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: tests/test_warp.py [hunk 1]
```diff
@@ -96,6 +98,18 @@
 
 
 class TestWarp(unittest.TestCase):
```
Comment (wyli) [inline tests/test_warp.py:100]: thanks, could you add a skip if decorator to check the itk installation in the unit tests? https://github.com/Project-MONAI/MONAI/blob/1516ca758090dd8ed2890b64554a77f681a0e224/tests/test_matshow3d.py#L27-L28

there are also some style issues https://github.com/Project-MONAI/MONAI/runs/4886923400?check_suite_focus=true

Comment (kate-sann5100) [reply tests/test_warp.py:100]: sure, I will work on that.

**Comment by kate-sann5100** [comment]
/black/black

File: tests/test_warp.py (hunk 1 above)
Comment (wyli) [reply tests/test_warp.py:99]: thanks, I fixed the downloading dep issue https://github.com/Project-MONAI/MONAI/pull/3692/commits/6b2251ecc915c8446c065c060c0fc1e21ec05a2a

**Review by wyli — APPROVED** [review]
this pr compares the interpolation and padding patterns against itk python. itk `ResampleImageFilter` is preferred over `WarpImageFilter` we should update when the official examples are upgraded. (https://github.com/InsightSoftwareConsortium/ITK/issues/809)

**Pull request merged**